src/util/result.h

```diff
@@ -8,16 +8,133 @@
 #include <attributes.h>
 #include <util/translation.h>
 
+#include <memory>
+#include <optional>
+#include <tuple>
 #include <variant>
+#include <vector>
 
 namespace util {
+namespace detail {
+//! Empty string list
+const std::vector<bilingual_str> EMPTY_LIST{};
 
+//! Helper to move elements from one container to another.
+template <typename T>
+void MoveElements(T& src, T& dest)
+{
+    dest.insert(dest.end(), std::make_move_iterator(src.begin()), std::make_move_iterator(src.end()));
+    src.clear();
+}
```

Comment:
Sorry for the ghost comment. I had realized after posting that this comment about moving it to span.h wasn't really applicable so I removed it, but apparently not timely enough. I agree with you.

The only further thought I had was that if there's no need to be generic (yet - and maybe never) it might be worth not doing that to simplify things a bit where possible? E.g. the below is easier to quickly understand imo:

```suggestion
//! Helper to move warnings and errors from one ErrorInfo to another.
void MoveMessages(std::vector<bilingual_str>& src, std::vector<bilingual_str>& dest)
{
    dest.insert(dest.end(), std::make_move_iterator(src.begin()), std::make_move_iterator(src.end()));
    src.clear();
}
```

No strong views either way though, so feel free to ignore. Just trying to minimize on what's already a lot of genericness.